scripts/events_module/relationship/pregnancy_events.py

```diff
@@ -54,110 +77,34 @@ def handle_having_kits(self, cat, clan):
         if not can_have_kits:
             return
 
-        # Roll to see if the cat will have kits.
-        if cat.mate:
-            chance = game.config["pregnancy"]["primary_chance_mated"]
+        # DETERMINE THE SECOND PARENT
+        # check if there is a cat in the clan for the second parent
+        second_parent, affair = self.get_second_parent(cat, game.settings['affair'])
+        second_parent_relation = None
+        if second_parent and second_parent.ID in cat.relationships:
+            second_parent_relation = cat.relationships[second_parent.ID]
+        elif second_parent:
+            second_parent_relation = Relationship(cat, second_parent)
+            cat.relationships[second_parent.ID] = second_parent_relation
+
+        # check if the second_parent is not none and if they also can have kits
+        if second_parent and not self.check_second_parent(cat, second_parent):
```

Comment:
Right now, not kits can occur if you have "no unknown fathers" set to True.  The else statement is reached if the second parent is none or the second parent is able to have kits, If unknown fathers is true, then the else statement will always early-return the function. 

I suggest changing this line to: 
`if second_parent:
     if not self.check_second_parent(cat, second_parent):
          return`

I suggest changing this t

Reply:
Don't know why the interface don't show it, I added your change.